samples/opengl/depthMapTriangulation/imageProcessor.cpp

```diff
@@ -0,0 +1,386 @@
+                float quad_right_down = quad_p(h, w);
+
+                // check the surroundings for p (quads) and d_p (vertices)
+
+                // if any of those contains a broken value propagate it, ...
+                if (quad_left_up == settings.broken_values || quad_left_down == settings.broken_values ||
+                    quad_right_up == settings.broken_values || quad_right_down == settings.broken_values)
+                {
+                    p(h, w) = settings.broken_values;
+                    continue;
+                }
+                // else choose the max
+                p(h, w) = std::max(std::max(std::max(quad_left_up, quad_left_down), quad_right_up), quad_right_down);
+
+                // --- find d_P and d_D per pixel ---
+
+                // initialize the surrounding pixels for calculating the normal
+                vec3 left  = image(h, w - 1);
+                vec3 right = image(h, w + 1);
+                vec3 up    = image(h - 1, w);
+                vec3 down  = image(h + 1, w);
+
+                vec3 viewing = -image(h, w);
+                viewing.normalize();
+                vec3 normal = cross(right - left, up - down);
+                normal.normalize();
+
+                // --- find d_D per pixel ---
+                float pixel_d_D = std::min(2.0f, std::max(0.5f, disparity(h, w) / median_disparity));
+
+                // --- p ---
+                p(h, w) *= pixel_d_D;
+            }
+        }
+    }
+}
+
+float Saiga::ImageProcessor::get_disparity(ImageView<float> depth_imageView, ImageView<float> disparity_imageView)
+{
+    // median disparity (broken pixels will not be used)
+    std::vector<float> disparities;
+
+    for (int h = 0; h < depth_imageView.height; ++h)
+    {
+        for (int w = 0; w < depth_imageView.width; ++w)
+        {
+            float depth = depth_imageView(h, w);
+
+            // check for broken depth
+            if (depth == settings.broken_values)
+            {
+                disparity_imageView(h, w) = 0;
+                continue;
+            }
+
+            disparity_imageView(h, w) = settings.cameraParameters.bf / depth;
+
+            disparities.push_back(disparity_imageView(h, w));
+        }
+    }
+
+    std::sort(disparities.begin(), disparities.end());
+    int disp_len = disparities.size();
+    if (disp_len > 0)
+    {
+        if (disp_len % 2 == 0)
+        {
+            return (disparities[(disp_len / 2) - 1] + disparities[disp_len / 2]) / 2;
+        }
+        else
+        {
+            return disparities[disp_len / 2];
+        }
+    }
+    return 0.0f;
+}
+
+void Saiga::ImageProcessor::use_hysteresis_threshold(ImageView<float> depth_image, ImageView<vec3> unprojected_image,
+                                                     ImageView<float> computed_values)
+{
+    int height = depth_image.height;
+    int width  = depth_image.width;
+    std::vector<bool> sure_edges_vec(height * width, false);
```

Comment:
```
TemplatedImage<bool> sure_edges(height,width);
sure_edges.makeZero();
```

Reply:
`TemplatedImage` still does not seem to support `bool` or am I using it wrong? With those exact lines I get an error

Reply:
Ah, you are right. You can leave it as it is.